discovery_engine/test/integration/search_test.dart

```diff
@@ -0,0 +1,240 @@
+      await engine.dispose();
+      await server.close();
+      await Directory(data.applicationDirectoryPath).delete(recursive: true);
+    });
+
+    test('a new search can be opened', () async {
+      final result = expectEvent<ActiveSearchRequestSucceeded>(
+        await engine.requestQuerySearch(searchTerm),
+      );
+
+      expect(result.items, isNotEmpty);
+      expect(result.search.searchBy, SearchBy.query);
+      expect(result.search.searchTerm, searchTerm);
+
+      expect(
+        expectEvent<ActiveSearchTermRequestSucceeded>(
+          await engine.getActiveSearchTerm(),
+        ).searchTerm,
+        equals(searchTerm),
+      );
+    });
+
+    test('a new search can not be opened twice', () async {
+      expectEvent<ActiveSearchRequestSucceeded>(
+        await engine.requestQuerySearch(searchTerm),
+      );
+
+      expect(
+        expectEvent<ActiveSearchRequestFailed>(
+          await engine.requestQuerySearch(searchTerm2),
+        ).reason,
+        equals(SearchFailureReason.openActiveSearch),
+      );
+    });
+
+    test('we can request more results until we ran out of results', () async {
+      final result = expectEvent<ActiveSearchRequestSucceeded>(
+        await engine.requestQuerySearch(searchTerm),
+      );
+
+      expect(result.items, isNotEmpty);
+      expect(result.search.searchBy, SearchBy.query);
+      expect(result.search.searchTerm, searchTerm);
+
+      final nextResult = expectEvent<NextActiveSearchBatchRequestSucceeded>(
+        await engine.requestNextActiveSearchBatch(),
+      );
+
+      expect(nextResult.items, isNot(equals(result.items)));
+      expect(nextResult.items, isNotEmpty);
+      expect(nextResult.search.searchBy, SearchBy.query);
+      expect(nextResult.search.searchTerm, searchTerm);
+
+      expect(
+        expectEvent<NextActiveSearchBatchRequestSucceeded>(
+          await engine.requestNextActiveSearchBatch(),
+        ).items,
+        isEmpty,
+      );
+    });
+
+    test('closing a search works', () async {
+      final result = expectEvent<ActiveSearchRequestSucceeded>(
+        await engine.requestQuerySearch(searchTerm),
+      );
+
+      expect(result.items, isNotEmpty);
+      expect(result.search.searchBy, SearchBy.query);
+      expect(result.search.searchTerm, searchTerm);
+
+      expectEvent<ActiveSearchClosedSucceeded>(
+        await engine.closeActiveSearch(),
+      );
+
+      expect(
+        expectEvent<ActiveSearchTermRequestFailed>(
+          await engine.getActiveSearchTerm(),
+        ).reason,
+        SearchFailureReason.noActiveSearch,
+      );
+
+      expect(
+        expectEvent<NextActiveSearchBatchRequestFailed>(
+          await engine.requestNextActiveSearchBatch(),
+        ).reason,
+        SearchFailureReason.noActiveSearch,
+      );
+
+      expect(
+        expectEvent<ActiveSearchClosedFailed>(
+          await engine.closeActiveSearch(),
+        ).reason,
+        SearchFailureReason.noActiveSearch,
+      );
```

Comment:
:thought_balloon: I guess returning `ActiveSearchClosedSucceeded` could also arguably be valid in this case... either way, it is good this case is tested.

Reply:
I had proposed that in the past. But it was decided that it should fail in this case (and similar cases).